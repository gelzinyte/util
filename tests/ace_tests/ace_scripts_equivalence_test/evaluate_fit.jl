using Pkg
Pkg.activate("/home/eg475/ace")

using IPFitting, ArgParse, ACE, LinearAlgebra, JSON3


function parse_commandline()
    s = ArgParseSettings()
    @add_arg_table s begin 
        "--param-fname", "-p"
            help="ace.json filename"
    end
	return parse_args(s)
end


function main()

	ip_fname = parse_commandline()["param-fname"]

    println("proper training set")
    actual_train_set = IPFitting.Data.read_xyz("/home/eg475/scripts/tests/files/tiny_gap.train_set.xyz", 
                                energy_key="dft_energy", force_key="dft_forces");


    println("Training set")
    train_set = IPFitting.Data.read_xyz("/data/eg475/heap_of_ch/train.dft.xyz", 
                                energy_key="dft_energy", force_key="dft_forces");


    println("Test set")
    test_set = IPFitting.Data.read_xyz("/data/eg475/heap_of_ch/test.dft.xyz", 
                                energy_key="dft_energy", force_key="dft_forces");

    IP = read_dict(load_dict(ip_fname)["IP"]);

    println("proper Train set")
    add_fits_serial!(IP, actual_train_set, fitkey="IP")
    rmse_, rmserel_ = rmse(actual_train_set; fitkey="IP");
    rmse_table(rmse_, rmserel_)


    println("Train set")
    add_fits_serial!(IP, train_set, fitkey="IP")
    rmse_, rmserel_ = rmse(train_set; fitkey="IP");
    rmse_table(rmse_, rmserel_)

    println("Test set")
    add_fits_serial!(IP, test_set, fitkey="IP")
    rmse_, rmserel_ = rmse(test_set; fitkey="IP");
    rmse_table(rmse_, rmserel_)

    d = load_dict(ip_fname);

	JSON3.pretty(d["info"])

end

main()
