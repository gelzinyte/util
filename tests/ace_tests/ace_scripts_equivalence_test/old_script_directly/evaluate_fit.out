proper training set
Reading in /home/eg475/scripts/tests/files/tiny_gap.train_set.xyz ...
Processing data ...
┌───────────────┬───────┬───────┬───────┬───────┬───────┐
│   config_type │ #cfgs │ #envs │    #E │    #F │    #V │
│        String │ Int64 │ Int64 │ Int64 │ Int64 │ Int64 │
├───────────────┼───────┼───────┼───────┼───────┼───────┤
│   methane_mol │    40 │   200 │    40 │   600 │     0 │
│  methane_rad1 │    40 │   160 │    40 │   480 │     0 │
│  methane_rad2 │    40 │   160 │    40 │   480 │     0 │
│  methane_rad3 │    40 │   160 │    40 │   480 │     0 │
│  methane_rad4 │    40 │   160 │    40 │   480 │     0 │
│ isolated_atom │     2 │     2 │     2 │     0 │     0 │
├───────────────┼───────┼───────┼───────┼───────┼───────┤
│         total │   202 │   842 │   202 │  2520 │     0 │
│       missing │     0 │     0 │     0 │     6 │  1818 │
└───────────────┴───────┴───────┴───────┴───────┴───────┘
Training set
Reading in /data/eg475/heap_of_ch/train.dft.xyz ...
Processing data ...
┌────────────────────┬───────┬───────┬───────┬───────┬───────┐
│        config_type │ #cfgs │ #envs │    #E │    #F │    #V │
│             String │ Int64 │ Int64 │ Int64 │ Int64 │ Int64 │
├────────────────────┼───────┼───────┼───────┼───────┼───────┤
│                 md │    49 │  1564 │    49 │  4692 │     0 │
│ normal_mode_sample │    23 │   756 │    23 │  2268 │     0 │
│          optimised │     1 │    29 │     1 │    87 │     0 │
│           opt_traj │    19 │   654 │    19 │  1962 │     0 │
├────────────────────┼───────┼───────┼───────┼───────┼───────┤
│              total │    92 │  3003 │    92 │  9009 │     0 │
│            missing │     0 │     0 │     0 │     0 │   828 │
└────────────────────┴───────┴───────┴───────┴───────┴───────┘
Test set
Reading in /data/eg475/heap_of_ch/test.dft.xyz ...
Processing data ...
┌────────────────────┬───────┬───────┬───────┬───────┬───────┐
│        config_type │ #cfgs │ #envs │    #E │    #F │    #V │
│             String │ Int64 │ Int64 │ Int64 │ Int64 │ Int64 │
├────────────────────┼───────┼───────┼───────┼───────┼───────┤
│                 md │    90 │  2937 │    90 │  8811 │     0 │
│ normal_mode_sample │    20 │   663 │    20 │  1989 │     0 │
│          optimised │     4 │   127 │     4 │   381 │     0 │
│           opt_traj │    57 │  1903 │    57 │  5709 │     0 │
├────────────────────┼───────┼───────┼───────┼───────┼───────┤
│              total │   171 │  5630 │   171 │ 16890 │     0 │
│            missing │     0 │     0 │     0 │     0 │  1539 │
└────────────────────┴───────┴───────┴───────┴───────┴───────┘
proper Train set
┌───────────────┬─────────┬──────────┬────────┬────────┐
│   config type │ E [meV] │ F [eV/A] │  F [%] │ V[meV] │
├───────────────┼─────────┼──────────┼────────┼────────┤
│ isolated_atom │   0.000 │      NaN │    NaN │    NaN │
│   methane_mol │   2.509 │    0.161 │ 22.287 │    NaN │
│  methane_rad1 │   4.672 │    0.166 │ 23.835 │    NaN │
│  methane_rad2 │   4.144 │    0.144 │ 18.927 │    NaN │
│  methane_rad3 │   5.056 │    0.158 │ 22.116 │    NaN │
│  methane_rad4 │   4.034 │    0.145 │ 19.992 │    NaN │
├───────────────┼─────────┼──────────┼────────┼────────┤
│           set │   4.154 │    0.155 │ 21.455 │    NaN │
└───────────────┴─────────┴──────────┴────────┴────────┘
Train set
┌────────────────────┬──────────┬──────────┬──────────┬────────┐
│        config type │  E [meV] │ F [eV/A] │    F [%] │ V[meV] │
├────────────────────┼──────────┼──────────┼──────────┼────────┤
│                 md │ 4688.138 │   25.842 │ 2304.992 │    NaN │
│ normal_mode_sample │ 4393.699 │   25.853 │ 2052.253 │    NaN │
│           opt_traj │ 4386.318 │   25.356 │ 8082.623 │    NaN │
│          optimised │ 3558.280 │   24.992 │ 8494.827 │    NaN │
├────────────────────┼──────────┼──────────┼──────────┼────────┤
│                set │ 4543.481 │   25.732 │ 2480.155 │    NaN │
└────────────────────┴──────────┴──────────┴──────────┴────────┘
Test set
┌────────────────────┬──────────┬──────────┬──────────┬────────┐
│        config type │  E [meV] │ F [eV/A] │    F [%] │ V[meV] │
├────────────────────┼──────────┼──────────┼──────────┼────────┤
│                 md │ 4362.926 │   24.452 │ 2303.323 │    NaN │
│ normal_mode_sample │ 4304.903 │   25.893 │ 2090.513 │    NaN │
│           opt_traj │ 4196.496 │   25.054 │ 8208.518 │    NaN │
│          optimised │ 4020.671 │   25.757 │ 8488.401 │    NaN │
├────────────────────┼──────────┼──────────┼──────────┼────────┤
│                set │ 4293.522 │   24.860 │ 2775.650 │    NaN │
└────────────────────┴──────────┴──────────┴──────────┴────────┘
{
                   "c": [
                          -0.010455070596146561,
                          0.01775521522932662,
                          0.08446512108288769,
                          -0.5074609659908339,
                          21.995603068776216,
                          -34.79808124353018,
                          -28.75736611295169,
                          -2.9723654303691145,
                          0,
                          0,
                          0,
                          0,
                          1.6734394274434985,
                          2.7013598008343904,
                          -1.148047024044734,
                          -1.0439976235599564,
                          1.7364989944125728,
                          -0.290343355000265,
                          -0.398414697786454,
                          1.6739790076240717,
                          -1.196019586518035,
                          0.5597675282053581,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          -2.5665607972007876,
                          0.8271034760893741,
                          -0.4613985593263031,
                          -1.106518828214955,
                          -0.11288399605360246,
                          0.06899996706426605,
                          1.599357679049445,
                          -0.6655084757122811,
                          1.5111877087290901,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          -0.17161749499959014,
                          -0.521611571785923,
                          -1.7366054061599814,
                          0.2256672251207879,
                          -0.798657869350609,
                          -2.3128501350580137,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          -0.11353365977405377,
                          -1.4712479157352645,
                          1.5579292909408498,
                          0,
                          0,
                          0,
                          -1.395387798007981,
                          -0.3604632035005251,
                          -0.7750475549593996,
                          -1.5719902604340967,
                          1.9928256961620257,
                          1.8903373568420305,
                          -0.9033799270460244,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0.7226991110079066,
                          2.379705527973367,
                          3.71385699622077,
                          3.1160860050403687,
                          0,
                          0,
                          0,
                          0,
                          0,
                          -0.906119655103008,
                          0,
                          0,
                          0.4151378949563053,
                          0.3015735912783275,
                          0.6113694488705824,
                          -2.227759006083146,
                          -2.010569143011933,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0.08141700660387936,
                          -1.8298642832414729,
                          0,
                          0,
                          0,
                          0,
                          0,
                          1.441176019949259,
                          -0.6727244075702403,
                          2.381538797921856,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0.7160388148259743,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0.27569592755637273,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0
                        ],
              "dbpath": "ace_old_script",
              "Ibasis": [
                          1,
                          2,
                          3,
                          4,
                          5,
                          6,
                          7,
                          8,
                          9,
                          10,
                          11,
                          12,
                          13,
                          14,
                          15,
                          16,
                          17,
                          18,
                          19,
                          20,
                          21,
                          22,
                          23,
                          24,
                          25,
                          26,
                          27,
                          28,
                          29,
                          30,
                          31,
                          32,
                          33,
                          34,
                          35,
                          36,
                          37,
                          38,
                          39,
                          40,
                          41,
                          42,
                          43,
                          44,
                          45,
                          46,
                          47,
                          48,
                          49,
                          50,
                          51,
                          52,
                          53,
                          54,
                          55,
                          56,
                          57,
                          58,
                          59,
                          60,
                          61,
                          62,
                          63,
                          64,
                          65,
                          66,
                          67,
                          68,
                          69,
                          70,
                          71,
                          72,
                          73,
                          74,
                          75,
                          76,
                          77,
                          78,
                          79,
                          80,
                          81,
                          82,
                          83,
                          84,
                          85,
                          86,
                          87,
                          88,
                          89,
                          90,
                          91,
                          92,
                          93,
                          94,
                          95,
                          96,
                          97,
                          98,
                          99,
                          100,
                          101,
                          102,
                          103,
                          104,
                          105,
                          106,
                          107,
                          108,
                          109,
                          110,
                          111,
                          112,
                          113,
                          114,
                          115,
                          116,
                          117,
                          118,
                          119,
                          120,
                          121,
                          122,
                          123,
                          124,
                          125,
                          126,
                          127,
                          128,
                          129,
                          130,
                          131,
                          132,
                          133,
                          134,
                          135,
                          136,
                          137,
                          138,
                          139,
                          140,
                          141,
                          142,
                          143,
                          144,
                          145,
                          146,
                          147,
                          148,
                          149,
                          150,
                          151,
                          152,
                          153,
                          154,
                          155,
                          156,
                          157,
                          158,
                          159,
                          160,
                          161,
                          162,
                          163,
                          164,
                          165,
                          166,
                          167,
                          168,
                          169,
                          170,
                          171,
                          172,
                          173,
                          174,
                          175,
                          176,
                          177,
                          178,
                          179,
                          180,
                          181,
                          182,
                          183,
                          184,
                          185,
                          186,
                          187,
                          188,
                          189,
                          190,
                          191,
                          192,
                          193,
                          194,
                          195,
                          196,
                          197,
                          198,
                          199,
                          200,
                          201,
                          202,
                          203,
                          204,
                          205,
                          206,
                          207,
                          208,
                          209,
                          210,
                          211,
                          212,
                          213,
                          214,
                          215,
                          216,
                          217,
                          218,
                          219,
                          220,
                          221,
                          222,
                          223,
                          224,
                          225,
                          226,
                          227,
                          228,
                          229,
                          230,
                          231,
                          232,
                          233,
                          234,
                          235,
                          236,
                          237,
                          238,
                          239,
                          240,
                          241,
                          242,
                          243,
                          244,
                          245,
                          246,
                          247,
                          248,
                          249,
                          250,
                          251,
                          252,
                          253,
                          254,
                          255,
                          256,
                          257,
                          258,
                          259,
                          260,
                          261,
                          262,
                          263,
                          264,
                          265,
                          266,
                          267,
                          268,
                          269,
                          270,
                          271,
                          272,
                          273,
                          274,
                          275,
                          276,
                          277,
                          278,
                          279,
                          280,
                          281,
                          282,
                          283,
                          284,
                          285,
                          286,
                          287,
                          288,
                          289,
                          290,
                          291,
                          292,
                          293,
                          294,
                          295,
                          296,
                          297,
                          298,
                          299,
                          300,
                          301,
                          302,
                          303,
                          304,
                          305,
                          306,
                          307,
                          308,
                          309,
                          310,
                          311,
                          312,
                          313,
                          314,
                          315,
                          316,
                          317,
                          318,
                          319,
                          320,
                          321,
                          322,
                          323,
                          324,
                          325,
                          326,
                          327,
                          328,
                          329,
                          330,
                          331,
                          332,
                          333,
                          334,
                          335,
                          336,
                          337,
                          338,
                          339,
                          340,
                          341,
                          342,
                          343,
                          344,
                          345,
                          346,
                          347,
                          348,
                          349,
                          350,
                          351,
                          352,
                          353,
                          354,
                          355,
                          356,
                          357,
                          358,
                          359,
                          360,
                          361,
                          362,
                          363,
                          364,
                          365,
                          366,
                          367,
                          368,
                          369,
                          370,
                          371,
                          372,
                          373,
                          374,
                          375,
                          376,
                          377,
                          378,
                          379,
                          380,
                          381,
                          382,
                          383,
                          384,
                          385,
                          386,
                          387,
                          388,
                          389,
                          390,
                          391,
                          392,
                          393,
                          394,
                          395,
                          396,
                          397,
                          398,
                          399,
                          400,
                          401,
                          402,
                          403,
                          404,
                          405,
                          406,
                          407,
                          408,
                          409,
                          410
                        ],
             "weights": {
                            "ignore": [],
                           "default": {
                                         "V": 1,
                                         "E": 1,
                                         "F": 1
                                      }
                        },
                 "p_1": 0,
               "kappa": 0,
              "solver": "lsqr",
        "juliaversion": "Julia Version 1.7.1\nCommit ac5cc99908 (2021-12-22 19:35 UTC)\nPlatform Info:\n  OS: Linux (x86_64-pc-linux-gnu)\n  CPU: Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz\n  WORD_SIZE: 64\n  LIBM: libopenlibm\n  LLVM: libLLVM-12.0.1 (ORCJIT, sandybridge)\nEnvironment:\n  JULIA_PROJECT = /home/eg475/ace\n  JULIA_NUM_THREADS = 16\n",
        "regularisers": [],
   "IPFitting_version": "na",
                  "E0": 0
}